azurerm/resource_arm_container_group.go

```diff
@@ -599,18 +620,33 @@ func expandContainerGroupContainers(d *schema.ResourceData) (*[]containerinstanc
 	return &containers, &containerGroupPorts, &containerGroupVolumes
 }
 
-func expandContainerEnvironmentVariables(input interface{}) *[]containerinstance.EnvironmentVariable {
+func expandContainerEnvironmentVariables(input interface{}, secure bool) *[]containerinstance.EnvironmentVariable {
+
 	envVars := input.(map[string]interface{})
 	output := make([]containerinstance.EnvironmentVariable, 0)
 
-	for k, v := range envVars {
-		ev := containerinstance.EnvironmentVariable{
-			Name:  utils.String(k),
-			Value: utils.String(v.(string)),
+	if secure == true {
+
+		for k, v := range envVars {
+			ev := containerinstance.EnvironmentVariable{
+				Name:        utils.String(k),
+				SecureValue: utils.String(v.(string)),
+			}
+
+			output = append(output, ev)
```

Comment:
Isn't the re-allocation happening too frequently?

Reply:
@metacpp - I am struggling with this one, most likely a misunderstanding on my part about how  this should work in Go.

Basically an environment variable and secure environment variable are the same thing, with the only difference being the name of the value property (Value vs. SecureValue).

In this code I am looping through each defined environment variable (both non-secure and secure), building an instance of the EnvironmentVariable type (Name and Value), and then adding this to a slice of EnvironmentVariables.

Here is the code with some notes added.

If you have any advice on how to proceed, I would appreciate them.

```
func expandContainerEnvironmentVariables(input interface{}, secure bool) *[]containerinstance.EnvironmentVariable {

	envVars := input.(map[string]interface{})
	output := make([]containerinstance.EnvironmentVariable, 0)

	// For each secure environment variable, build the instance and add to output (slice)
	if secure == true {
		for k, v := range envVars {
			ev := containerinstance.EnvironmentVariable{
				Name:        utils.String(k),
				SecureValue: utils.String(v.(string)),
			}
			output = append(output, ev)
		}

	} else {
		// For each non-secure environment variable, build the instance and add to output (slice)
		for k, v := range envVars {
			ev := containerinstance.EnvironmentVariable{
				Name:  utils.String(k),
				Value: utils.String(v.(string)),
			}
			output = append(output, ev)
		}
	}
	// Return slice which contains both secure and non-secure environment variables
	return &output
}
```